### PR TITLE
update the container-service sdk from 2020-02-01 to 2021-03-01

### LINKED ISSUE #1407: Add AAD Support for AKS
/kind feature
/area managedclusters

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]
You can integrate Azure Active Directory with an AKS cluster and use that to authorise users who access the cluster. We should add support for this in CAPZ

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### LINKED ISSUE #1418: Update containerservice sdk API version from 2020-02-01 to 2021-05-01
/kind feature
/area managedclusters

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]
Update the container service sdk from 2020-02-01 to the latest as the existing api is outdated and does not have the latest changes.

eg: The existing api has support only for legacy AAD integration with AKS. The latest API model provides support for the new style of integration of AAD with AKS. 

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by LochanRn**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Updates the existing containerService sdk to the latest version i.e 2021-03-01.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1418  

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
container-service api update
```

### CODE REVIEW

**Comment by LochanRn** [comment]
This is just an initial commit to review changes/modifications done after changing the containerservice sdk.

**Review by devigned — CHANGES_REQUESTED** [review]
The update of the API version looks good. I only question the `VMSizeTypes` to `*string` change and what it means about the behavior of the `PUT` request to Azure.

The change can be seen by comparing: [API version 2021-02-01](https://github.com/Azure/azure-sdk-for-go/blob/8cce9bda0fa6d8c5418c65b2f890cb382197a18d/services/containerservice/mgmt/2021-02-01/containerservice/models.go#L1226) and [API version 2021-03-01](https://github.com/Azure/azure-sdk-for-go/blob/8cce9bda0fa6d8c5418c65b2f890cb382197a18d/services/containerservice/mgmt/2021-03-01/containerservice/models.go#L1270).

One thing to consider. Perhaps, this should be it's own PR, rather than combining this with the AAD support as the name of the PR implies, `[WIP] aad support for aks`.

File: azure/services/agentpools/agentpools.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ func (s *Service) Reconcile(ctx context.Context, spec interface{}) error {
 
 	profile := containerservice.AgentPool{
 		ManagedClusterAgentPoolProfileProperties: &containerservice.ManagedClusterAgentPoolProfileProperties{
-			VMSize:              containerservice.VMSizeTypes(agentPoolSpec.SKU),
+			VMSize:              &agentPoolSpec.SKU,
```
[the remark this answers is not in the document]
Comment (alexeldeib) [reply azure/services/agentpools/agentpools.go:55]: You’d be correct :) 

https://github.com/Azure/azure-rest-api-specs/pull/13763#issuecomment-824171240

[the remark this answers is not in the document]
Comment (devigned) [reply azure/services/agentpools/agentpools.go:55]: Thank you for the confirmation, @alexeldeib!

**Comment by CecileRobertMichon** [comment]
@LochanRn please amend the commit to accurately describe what this is doing (it's not actually adding support for AAD)

**Comment by CecileRobertMichon** [comment]
I tried creating an aks cluster from this changeset, getting:

```
[manager] E0607 22:33:26.804946       7 controller.go:302] controller-runtime/manager/controller/azuremanagedcontrolplane "msg"="Reconciler error" "error"="error creating AzureManagedControlPlane default/aks-template: failed to reconcile managed cluster: failed to reconcile managed cluster aks-template: failed to create managed cluster, failed to begin operation: containerservice.ManagedClustersClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: Code=\"MustDefineAtLeastOneSystemPool\" Message=\"Must define at least one system pool.\"" "name"="aks-template" "namespace"="default" "reconciler group"="infrastructure.cluster.x-k8s.io" "reconciler kind"="AzureManagedControlPlane" 
```

/hold

**Comment by CecileRobertMichon** [comment]
@alexeldeib can you validate that setting the agent pool to mode System for now is appropriate to mitigate the error until #1376 is implemented?

**Comment by alexeldeib** [comment]
I think we need to implement #1376 in tandem with the SDK update tbh, or else deletion behavior will get pretty messy. @LochanRn reached out on slack and sounds like will address #1376 + SDK update. I think the AAD functionality would be good in a follow up PR after that.

**Comment by LochanRn** [comment]
closing this PR as it is fixed in #1475

**Pull request closed without merging**